jscomp/bsb/bsb_theme_init.ml

```diff
@@ -26,138 +26,139 @@
-  match x with 
-  | File (name,content)  -> 
+let rec process_theme_aux env cwd (x : OCamlRes.Res.node) =
+  match x with
+  | File (name,content)  ->
     Ext_io.write_file (cwd // name) (replace content env)
-  | Dir (current, nodes) -> 
+  | Dir (current, nodes) ->
     Unix.mkdir (cwd//current) 0o777;
     List.iter (fun x -> process_theme_aux env (cwd//current) x ) nodes
 
 let list_themes () =
   Format.fprintf Format.std_formatter "Available themes: @.";
-  Bsb_templates.root 
+  Bsb_templates.root
   |>
   List.iter (fun (x : OCamlRes.Res.node)  ->
-      match  x with 
-      | Dir (x, _) -> 
-        Format.fprintf Format.std_formatter "%s@." x 
+      match  x with
+      | Dir (x, _) ->
+        Format.fprintf Format.std_formatter "%s@." x
 
       | _ -> ()
-    ) 
+    )
 
-(* @raise [Not_found] *)  
-let process_themes env theme proj_dir (themes : OCamlRes.Res.node list ) = 
-  match List.find (fun (x : OCamlRes.Res.node) -> 
-      match  x with 
+(* @raise [Not_found] *)
+let process_themes env theme proj_dir (themes : OCamlRes.Res.node list ) =
+  match List.find (fun (x : OCamlRes.Res.node) ->
+      match  x with
       | Dir (dir, _) -> dir = theme
-      | File _ -> false 
-    ) themes  with 
-  | exception Not_found -> 
+      | File _ -> false
+    ) themes  with
+  | exception Not_found ->
     list_themes ();
     raise (Arg.Bad( "theme " ^ theme ^ " not found")  )
-  | Dir(_theme, nodes ) -> 
+  | Dir(_theme, nodes ) ->
     List.iter (fun node -> process_theme_aux env proj_dir node ) nodes
-  | _ -> assert false  
+  | _ -> assert false
 
 (** TODO: run npm link *)
-let init_sample_project ~cwd ~theme name = 
-  let env = String_hashtbl.create 0 in 
-  List.iter (fun (k,v) -> String_hashtbl.add env k v  ) [  
+let init_sample_project ~cwd ~theme name =
+  let env = String_hashtbl.create 0 in
+  List.iter (fun (k,v) -> String_hashtbl.add env k v  ) [
     "proj-version", "0.1.0";
     "bs-version", Bs_version.version;
     "bsb" , Filename.current_dir_name // "node_modules" // ".bin" // "bsb"
   ];
-  let action = fun _ -> 
-    process_themes env  theme Filename.current_dir_name Bsb_templates.root;         
-    run_npm_link cwd name
-  in   
-  begin match name with 
-    | "." -> 
+  let action ?(theme=theme) () =
+    process_themes env theme Filename.current_dir_name Bsb_templates.root;
+    (* Don't symlink `bs-platform` if already installed locally. *)
+    if not (Sys.file_exists (cwd // "node_modules" // "bs-platform"))
+      then run_npm_link cwd name else ()
+  in
+  begin match name with
+    | "." ->
       let name = Filename.basename cwd in
-      if Ext_namespace.is_valid_npm_package_name name then 
-        begin 
+      if Ext_namespace.is_valid_npm_package_name name then
+        begin
+          (* Ignores theme requests if a `package.json` is found and uses interop theme. *)
+          let theme_to_use = if Sys.file_exists (cwd // "package.json")
+            then "interop" else theme in
```

Comment:
can we detect between `bsb -init .` and `bsb -init . -theme x`, we probably should fail in the last case unless it happens to be interop

Reply:
hmm maybe, it looks like when we don't specify a theme we default to `basic`, so I guess we could fail when `bsb -init .` is called with any theme besides `basic` (and possibly `interop`)?

Is there a better way to check if the theme flag is set?